qiskit_ibm/runtime/ibm_runtime_service.py

```diff
@@ -231,21 +234,27 @@ def run(
             inputs.validate()
             inputs = vars(inputs)
 
+        colon_index = image.find(":")
+        if image and (image.count(":") != 1 or colon_index == 0 or colon_index == len(image) - 1):
```

Comment:
```suggestion
        if image and not re.match(r"\w+:\w+$", image):
```

This validation can be done more cleanly with a regular expression (don't forget to import `re`)

Reply:
`r"\w+:\w+$"`won't match for example `"us.icr.io:latest"`. Should we be concerned about the performance of regex?

Reply:
Yeah I guess it's more complicated than that, maybe something like `[a-z0-9]+(?:[._-]{1,2}[a-z0-9]+)*:*\w+[\w\.-]*$`? Don't quote me on this though. Anyway I still think regex is the way to go, since it's the standard for pattern matching. I'm not concerned about performance as regex is very fast.

I do have a separate question - normally if you don't specify an image tag, `latest` is used. But I'm assuming the server doesn't support specifying an image name without tag?

Reply:
The server have simple format validation, which requires image name and tag

Reply:
Are you suggesting we should just forgo validation here and let the server handle it?

Reply:
The server only checks for ":" in the string. I think we can mirror that to save an API call in that case.